sdk/file_sdk/XMPFileFormat/xmp_helper.h

```diff
@@ -0,0 +1,47 @@
+#ifndef XMP_HELPER_H
+#define XMP_HELPER_H
+
+#define TXMP_STRING_TYPE std::string
+#define XMP_INCLUDE_XMPFILES 1
+
+#include "xmpio_over_istream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+#include "tag.h"
+
+using std::string;
+using std::vector;
+using std::shared_ptr;
+
+namespace {
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
+const XMP_StringPtr kMsipLabelNamespace = "http://www.microsoft.com/msip/label/1.0/";
+const XMP_StringPtr kMsipLabelExtendedNamespace = "http://www.microsoft.com/msip/label/extended/1.0/";
+const XMP_StringPtr kLabelsArrayName = "labels";
+const XMP_StringPtr kExtendedStructName = "extended";
+}
+
+namespace mip {
+namespace file {
+
+class XMPHelper {

```

Comment:
all this can be static on singleton.

Reply:
static (private constructor) or singleton?

---
In reply to: [136713002](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r136713002) [](ancestors = 136713002)